python/ray/tune/execution/trial_runner.py

```diff
@@ -1083,8 +1086,12 @@ def add_trial(self, trial: Trial):
         """
         # If the config map has had all the references replaced,
         # resolve them before adding the trial.
-        if self._replaced_ref_map:
-            resolve_placeholders(trial.config, self._replaced_ref_map)
+        if self._placeholder_resolvers:
+            # Construct the full experiment spec for resolution.
+            spec = self._spec or {}
```

Comment:
Why are we updating the whole spec and not just the config?

Reply:
Or I guess, which other elements from the spec do we need? Seems like we're not using the rest of the spec here. If that's the case can we just go back to resolving `trial.config` - we're overwriting the `spec["config"]` with it anyways.

Reply:
I don't want to use spec either man. but Function taking spec is our public API ... the following is in our documentation.
```
"beta": tune.sample_from(lambda spec: spec.config.alpha * np.random.normal()),
```